### PR TITLE
Fix test, upgrade to go 1.19

### PR DESCRIPTION
**Pull request by ctrl-alt-lulz**

### CODE REVIEW

**Comment by treeder** [comment]
Thanks for this! I will try to review it today.

File: accounts/accounts.go [hunk 1]
```diff
@@ -23,29 +30,42 @@ func ParsePrivateKey(pkHex string) (*Account, error) {
...
 	return &Account{
 		key: key,
 	}, nil
 }
 
-type Account struct {
-	key *ecdsa.PrivateKey
-}
-
-func (a *Account) Key() *ecdsa.PrivateKey {
```
Comment (treeder) [inline accounts/accounts.go:37]: Breaking backwards compatibility.

File: client/client.go [hunk 2]
```diff
@@ -1,37 +1,43 @@
...
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"log"
 	"math/big"
 	"sync/atomic"
 
+	zlog "github.com/rs/zerolog/log"
+	web3_types "github.com/zeus-fyi/gochain/web3/types"
```
Comment (treeder) [inline client/client.go:13]: Not the right package name.

File: client/client.go [hunk 3]
```diff
@@ -72,10 +79,11 @@ func (c *client) Close() {
 	c.r.Close()
 }
 
-func (c *client) Call(ctx context.Context, msg CallMsg) ([]byte, error) {
+func (c *client) Call(ctx context.Context, msg web3_types.CallMsg) ([]byte, error) {
```
Comment (treeder) [inline client/client.go:82]: Breaking backwards compatibility. Along with all the other similar changes.

File: client/client.go [hunk 4]
```diff
@@ -137,24 +153,30 @@ func (c *client) GetID(ctx context.Context) (*ID, error) {
...
 		if e.Error != nil {
+			zlog.Err(e.Error).Msg("GetID: BatchCallContext")
 			log.Printf("Method %q failed: %v\n", e.Method, e.Error)
 		}
 	}
 	netID := new(big.Int)
 	if _, ok := netID.SetString(netIDStr, 10); !ok {
-		return nil, fmt.Errorf("invalid net_version result %q", netIDStr)
+		err := fmt.Errorf("invalid net_version result %q", netIDStr)
+		zlog.Err(err).Msg("GetID: netID.SetString(netIDStr, 10)")
```
Comment (treeder) [inline client/client.go:163]: Shouldn't be writing log messages in here, caller can log if they want. Same goes for probably all the logging you added.

**Review by treeder — CHANGES_REQUESTED** [review]
PR is too big to review in one shot, but from what I've seen, it's not backwards compatible. The changes would have to be really important to justify all the breaking changes.

File: client/client.go (hunk 4 above)
Comment (treeder) [reply client/client.go:163]: All the logging that is just prior to returning an error anyways.

**Comment by ctrl-alt-lulz** [comment]
@treeder, yeah figured it would probably not be compatible anymore after I started making major changes. I'll just maintain the alternative config in my repo. Reorganized the code to make it easier to use via API

**Pull request closed without merging**